GoogleContainerTools/skaffold
# add privacy notice and command to set update check false

Adding a config for turning off update check for skaffold

Test Notes:

1. Build a non dirty version with version set to < 0.37.0
```
GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go install -tags "osusergo netgo static_build" -ldflags " -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.version=v0.30.0 -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.buildDate=2019-08-30T10:29:55Z -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.gitCommit=13a5d595d7bcc760d00460e80dd78fa0f676d1ac -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.gitTreeState=dirty  -extldflags \"-static\"" -gcflags "all=-trimpath=/usr/local/google/home/tejaldesai/workspace/skaffold1" -asmflags "all=-trimpath=/usr/local/google/home/tejaldesai/workspace/skaffold1" github.com/GoogleContainerTools/skaffold/cmd/skaffold
```

2. Verify skaffold version is < 0.37.0 and `GitTreeState` is non-dirty to by-pass this condition https://github.com/GoogleContainerTools/skaffold/blob/master///pkg/skaffold/update/update.go#L42
```
$ skaffold version -v=debug
INFO[0000] Skaffold &{Version:t0.40.0 ConfigVersion:skaffold/v1beta14 GitVersion: GitCommit:13a5d595d7bcc760d00460e80dd78fa0f676d1ac GitTreeState:test BuildDate:2019-08-30T10:29:55Z GoVersion:go1.12.5 Compiler:gc Platform:linux/amd64} 
v0.30.0
```

3. run skaffold deploy in examples/getting-started and verify we see the message for newer version
```
$ cd examples/geting-started
$ skaffold deploy
Starting deploy...
kubectl client version: 1.11+
kubectl version 1.12.0 or greater is recommended for use with Skaffold
 - pod/getting-started configured
Deploy complete in 949.223543ms
There is a new version (0.37.0) of Skaffold available. Download it at https://storage.googleapis.com/skaffold/releases/latest/skaffold-linux-amd64
```

4. Now disable the update-check config via command line 
```
$skaffold config set -g update-check false
set global value update-check to false
```

5. Run skaffold deploy again and make sure there is no message for newer skaffold version.
```
tejaldesai@@getting-started (add_privacy_policy)$ skaffold deploy 
Starting deploy...
kubectl client version: 1.11+
kubectl version 1.12.0 or greater is recommended for use with Skaffold
 - pod/getting-started configured
Deploy complete in 1.170797454s
```

You can test all scenarios in the unit test.
